src/odemis/driver/zeiss.py

```diff
@@ -483,20 +518,37 @@ def terminate(self):
             self._pos_poll.cancel()
             self._pos_poll = None
 
-    def _updatePosition(self, raw_pos=None):
+    def _getStagePosition(self) -> Tuple[Dict[str, float], bool]:
         """
-        update the position VA
-        raw_pos (dict str -> float): the position in mm (as received from the SEM)
+        Read the stage position, using either GetStagePosition of GetStagePosition6 where the first supports 3
+        linear axes and the latter 6 axes including rx and rz rotational axes.
+        :return position dict str -> float: (in m/rad) with the axes supported in .axes 
+                is_moving (bool)
         """
-        if raw_pos is None:
-            x, y, z, _ = self.parent.GetStagePosition()
+        pos = {}
+
+        if len(self.axes) > 3:
```

Comment:
Now it's actually possible to disable any of the axes, including "x", "y", and "z". So you cannot rely just on the number of axes. You should check if rx, rz, or m are in the axes, and in that case use the 6 version.